### PR TITLE
chore(java_indexer): downgrade doc reference failure logging

### PR DESCRIPTION
**Pull request by shahms**

### CODE REVIEW

File: kythe/java/com/google/devtools/kythe/analyzers/java/KytheDocTreeScanner.java [hunk 1]
```diff
@@ -109,7 +109,8 @@ public Void visitReference(ReferenceTree tree, DCDocComment doc) {
     try {
       sym = (Symbol) trees.getElement(getCurrentPath());
     } catch (Throwable e) {
-      logger.atWarning().withCause(e).log("Failed to resolve documentation reference: %s", tree);
+      logger.atInfo().log("Failed to resolve documentation reference: %s\n%s", tree, e);
```
Comment (jaysachs) [inline kythe/java/com/google/devtools/kythe/analyzers/java/KytheDocTreeScanner.java:112]: Do we even need this? How is it useful outside of the case where we turn on fine logging to get the full trace?

Comment (shahms) [reply kythe/java/com/google/devtools/kythe/analyzers/java/KytheDocTreeScanner.java:112]: I think it's still useful to know about; it's something I expect people will file bugs about so is good to have a trace in the logs.

Comment (jaysachs) [reply kythe/java/com/google/devtools/kythe/analyzers/java/KytheDocTreeScanner.java:112]: Should we add a diagnostic instead?

Comment (shahms) [reply kythe/java/com/google/devtools/kythe/analyzers/java/KytheDocTreeScanner.java:112]: Yeah, good call. Just a bit more involved.

**Review by jaysachs — APPROVED** [review]

**Pull request merged**